File: Two Pointers/283. Move Zeroes/283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        for i in range(len(nums)-1, -1, -1):
            k = -10
            if nums[i] == 0:
                k = i
                for j in range(k, len(nums) - 1):
                    nums[j], nums[j + 1] = nums[j + 1], nums[j]

File: Two Pointers/283. Move Zeroes/283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    for (let i = nums.length - 1; i >= 0; i--) {
        if (nums[i] === 0) {
            // Перемещаем ноль в конец, сдвигая элементы
            for (let j = i; j < nums.length - 1; j++) {
                // Меняем местами текущий элемент со следующим
                let temp = nums[j];
                nums[j] = nums[j + 1];
                nums[j + 1] = temp;
            }
        }
    }
};